provider/src/index.ts

```diff
@@ -26,15 +29,55 @@ export default {
     env: Env,
     ctx: ExecutionContext
   ): Promise<Response> {
+    // Handling CORS
+    if (request.method === 'OPTIONS') {
+      return new Response('', {
+        headers: {
+          'access-control-allow-methods': 'POST',
+          'access-control-allow-headers': 'content-type',
+          'access-control-max-age': '1800',
+          'access-control-allow-origin': request.headers.get('Origin'),
+          vary: 'Origin',
+          'access-control-allow-credentials': 'true',
+        },
+      })
+    }
+
+    // Reject requests that are not POST
+    if (request.method !== 'POST') {
+      return new Response('Method not supported', { status: 400 })
+    }
+
     const url = new URL(request.url)
     const { pathname } = url
     const [_, networkId] = pathname.match(/\/([0-9]*)/)
+
+    // Missing network
     if (!networkId) {
       return new Response('Bad Request', { status: 400 })
     }
+
+    // Network not supported
     if (!supportedNetworks[networkId]) {
       return new Response('Network not supported', { status: 404 })
     }
-    return fetch(supportedNetworks[networkId], request)
+
+    // Make JSON RPC request
+    const response = await fetch(supportedNetworks[networkId], {
+      method: 'POST',
+      body: request.body,
+      headers: new Headers({
+        Accept: '*/*',
+        Origin: 'https://rpc.unlock-protocol.com/', // required to add this to allowlists
+      }),
+    })
+    const json = await response.json()
+
+    return new Response(JSON.stringify(json), {
+      headers: {
+        'content-type': 'application/json',
+        'access-control-allow-origin': request.headers.get('Origin'),
+      },
+    })
```

Comment:
> you can clone the response directly instead of parsing to json, stringifying, etc here.

I tried but that does not work :( Either you pass the full `response` object or you have to reconstruct a full response.